src/integration/java/org/humancellatlas/ingest/biomaterial/BiomaterialControllerTest.java

```diff
@@ -0,0 +1,143 @@
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.NO_CONTENT.value());
+
+        verify(validationStateChangeService, times(1)).changeValidationState(any(), any(), eq(ValidationState.DRAFT));
+    }
+
+    @Test
+    public void testSaveInputToProcessTriggersValidationStateToDraft() throws Exception {
+        // given
+        Process process = new Process(UUID.randomUUID());
+        processRepository.save(process);
+
+        // and
+        Biomaterial biomaterial = new Biomaterial(null);
+        biomaterialRepository.save(biomaterial);
+
+        // send post request
+        MvcResult result = webApp.perform(post("/biomaterials/{biomaterialId}/inputToProcesses/", biomaterial.getId())
+                .contentType("text/uri-list")
+                .content(ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process.getId()))
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.NO_CONTENT.value());
+
+        verify(validationStateChangeService, times(1)).changeValidationState(any(), any(), eq(ValidationState.DRAFT));
+
+    }
+
+    @Test
+    public void testDeleteDerivedByProcessTriggersValidationStateToDraft() throws Exception {
+        // given
+        Process process = new Process(UUID.randomUUID());
+        processRepository.save(process);
+
+        // and
+        Biomaterial biomaterial = new Biomaterial(null);
+        biomaterial.addAsDerivedByProcess(process);
+        biomaterialRepository.save(biomaterial);
+
+        // send delete request
+        MvcResult result = webApp.perform(delete("/biomaterials/{biomaterialId}/derivedByProcesses/{processId}", biomaterial.getId(), process.getId()))
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.NO_CONTENT.value());
+
+        verify(validationStateChangeService, times(1)).changeValidationState(any(), any(), eq(ValidationState.DRAFT));
+    }
+
+    @Test
+    public void testDerivedByProcessTriggersValidationStateToDraft() throws Exception {
+        // given
+        Process process = new Process(UUID.randomUUID());
+        processRepository.save(process);
+
+        // and
+        Biomaterial biomaterial = new Biomaterial(null);
+        biomaterialRepository.save(biomaterial);
+
+        // send post request
+        MvcResult result = webApp.perform(post("/biomaterials/{biomaterialId}/derivedByProcesses/", biomaterial.getId())
+                .contentType("text/uri-list")
+                .content(ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process.getId()))
+                .andReturn();
```

Comment:
Instead returning and then checking the status, you could use
```suggestion
                .andExpect(MockMvcResultMatchers.status(). isNoContent())
```

See: 

- [andExpect](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/web/servlet/ResultActions.html#andExpect-org.springframework.test.web.servlet.ResultMatcher-) web result checker action.
- [other result matchers](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/web/servlet/result/MockMvcResultMatchers.html)

Reply:
good idea! applied, thanks!

Reply:
yep. Using the fluent API. I suggested the very same thing about a month or so ago to Yusra. 😂